File: leetcode/POTD/2025-02-19/solution.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        res = []

        def helper(s: str) -> bool:
            return all(s[i] != s[i - 1] for i in range(1, len(s)))

        def gen_permut(s: str):
            if len(s) == n:
                if helper(s):
                    res.append(s)
                return
            for ch in "abc":
                gen_permut(s + ch)

        gen_permut("")
        return res[k - 1] if len(res) >= k else ""

File: leetcode/POTD/2025-02-19/solution.ts
function getHappyString(n: number, k: number): string {
	let res: string[] = []

	const helper = (s: string): boolean => {
		for (let i = 1; i < s.length; i++) {
			if (s[i] === s[i - 1]) return false
		}
		return true
	}

	const genPermut = (s: string): void => {
		if (s.length === n) {
			if (helper(s)) res.push(s)
			return
		}
		for (let ch of ["a", "b", "c"]) {
			genPermut(s + ch)
		}
	}

	genPermut("")
	return res.length < k ? "" : res[k - 1]
}
